sections/video.liquid

```diff
@@ -31,32 +47,18 @@
     <noscript>
       <div
         class="video-section__media"
-        {% if section.settings.cover_image != blank %}
-          style="padding-bottom: {{ 1 | divided_by: section.settings.cover_image.aspect_ratio | times: 100 }}%;"
+        {% if poster != null %}
+          style="--ratio-percent: {{ 1 | divided_by: poster.aspect_ratio | times: 100 }}%;"
```

Comment:
For the border width issue, it looks like a simple `padding-bottom: var(--ratio-percent);` is all that's needed for fullwidth. Subtracting the border width when we don't have left/right borders was causing that discrepancy. 

As far as basing the aspect ratio off the video rather than the cover image, we don't think we should do this right now. As implemented the empty space behavior is equivalent to the youtube and vimeo players when you choose to use a cover image of a different aspect ratio. It could be something we could consider changing in the future, but we'd want to make the change for all video types and this could be a considered visually breaking change. 

cc @melissaperreault in case you have anything to add or clarify